echo-scheduler/src/main/groovy/com/netflix/spinnaker/echo/scheduler/actions/pipeline/PipelineConfigsPollingJob.groovy

```diff
@@ -169,24 +176,69 @@ class PipelineConfigsPollingJob implements Job {
     registry.gauge("echo.triggers.sync.addCount").set(addCount)
   }
 
+  boolean storeTrigger(Trigger pipelineTrigger) {
+    return storeTrigger(pipelineTrigger, null)
+  }
+
   /**
-   * Register a new scheduler trigger given a pipeline trigger
+   * Register a new (or re-register existing) scheduler trigger given a pipeline trigger
    * @param pipelineTrigger
    */
-  boolean registerNewTrigger(Trigger pipelineTrigger) {
-    boolean isSuccess = false
+  boolean storeTrigger(Trigger pipelineTrigger, TriggerKey triggerKey) {
+    org.quartz.Trigger trigger
 
     try {
-      scheduler.scheduleJob(TriggerConverter.toQuartzTrigger(pipelineTrigger, timeZoneId))
-      isSuccess = true
+      trigger = TriggerConverter.toQuartzTrigger(pipelineTrigger, timeZoneId)
     } catch (InvalidCronExpressionException e) {
-      log.error("Failed to create a new trigger: id: ${pipelineTrigger.id} for pipeline: ${pipelineTrigger.parent.application}:${pipelineTrigger.parent.name} (${pipelineTrigger.parent.id}). " +
-        "The CRON expression '${pipelineTrigger.cronExpression}' is not valid")
-    } catch (Exception e) {
-      log.error("Failed to create a new trigger: id: ${pipelineTrigger.id} for pipeline: ${pipelineTrigger.parent.id}. " +
-        "The CRON expression '${pipelineTrigger.cronExpression}'", e)
+      log.warn("Failed to create a new trigger: id: ${pipelineTrigger.id} for pipeline: ${pipelineTrigger.parent.application}:${pipelineTrigger.parent.name} (${pipelineTrigger.parent.id}). " +
+        "The CRON expression '${pipelineTrigger.cronExpression}' is not valid", e)
+
+      return false
+    }
+
+    // It's possible the user has created a trigger that will never fire (e.g. a trigger in the past)
+    // That's ok, just don't even bother creating it
+    try {
+      boolean willTriggerFire = willTriggerFire(trigger)
+
+      if (!willTriggerFire) {
+        throw new SchedulerException("Trigger is in the past")
```

Comment:
Going to get a noisy stacktrace when this fires ... perhaps that's ok ... :shrug:

Reply:
yes, at least it's not going generate a failure metric.. this code (even though i wrote it... is giving me a headache...)
for a while, i wanted to add some logic that would not spit out an error for the same trigger over and over again (since we sync every ~60s) maybe one day..